src/components/organisms/AssetActions/Consume.tsx

```diff
@@ -167,7 +167,11 @@ export default function Consume({
         </div>
         <div className={styles.pricewrapper}>
           <Price price={price} conversion />
-          {!isInPurgatory && <PurchaseButton />}
+          {parseInt(file?.contentLength) === 0 ? (
```

Comment:
`file` here refers to the file metadata object within the DDO, which is only set upon publish, retrieved when file is added. So this check here would only kick in if file has been published with `contentLength: 0` which is very unlikely

Reply:
I think we can use this provide POST method (https://docs.oceanprotocol.com/references/provider/#apiv1servicesfileinfo) to get the file Content-Type and Content-Length using the assed did